app/src/test/java/cm/aptoide/pt/app/AppViewManagerTest.java

```diff
@@ -0,0 +1,717 @@
+
+    MinimalAdRequestResult minimalAdRequestResult =
+        new MinimalAdRequestResult(AppsList.Error.GENERIC);
+    AppsList appsList = new AppsList(Collections.emptyList(), false, 0);
+
+    //When the presenters asks for the Similar Apps bundle
+    //And the result of the Ad request returns no Ad and the result of the Recommended Apps return a list of SimilarApps (empty or not)
+    when(adsManager.loadAd("anyString", keywords)).thenReturn(Single.just(minimalAdRequestResult));
+    when(appCenter.loadRecommendedApps(limit, "anyString")).thenReturn(Single.just(appsList));
+
+    SimilarAppsViewModel similarAppsViewModel =
+        appViewManager.loadSimilarApps("anyString", keywords)
+            .toBlocking()
+            .value();
+
+    //Then a request of recommendedApps and an Ad should be done
+    verify(appCenter).loadRecommendedApps(limit, "anyString");
+    verify(adsManager).loadAd("anyString", keywords);
+
+    //And a SimilarAppsViewModel should be returned with no Ad, a list of similarApps, no loading, no error in recommendedApps, a error in of an Ad, and saying that there was an error
+    Assert.assertNull(similarAppsViewModel.getAd());
+    Assert.assertEquals(Collections.emptyList(), similarAppsViewModel.getRecommendedApps());
+    Assert.assertEquals(false, similarAppsViewModel.isLoading());
+    Assert.assertEquals(true, similarAppsViewModel.hasError());
+    Assert.assertEquals(true, similarAppsViewModel.hasAdError());
+    Assert.assertEquals(false, similarAppsViewModel.hasRecommendedAppsError());
+  }
+
+  @Test public void loadAdsFromAppViewTest() {
+    //Cache App (Test preparation)
+    DetailedApp detailedApp =
+        new DetailedApp((long) 1, "any", "anyString", (long) 1, "any", "any", "any", "any", true,
+            null, null, null, null, null, (long) 1, null, null, null, 1, null, null, store, null,
+            appStats, null, null, null, true, true, null);
+    MinimalAd minimalAd =
+        new MinimalAd("anyString", (long) 1, "", "", "", (long) 1, (long) 1, "", "", "", "", 1, 1,
+            (long) 1);
+    DetailedAppRequestResult detailedAppRequestResult = new DetailedAppRequestResult(detailedApp);
+
+    AppViewConfiguration appViewConfiguration =
+        new AppViewConfiguration((long) 1, "anyString", "anyString", "", null, null, "", "", 0.0,
+            "", "");
+
+    appViewManager =
+        new AppViewManager(installManager, downloadFactory, appCenter, reviewsManager, adsManager,
+            storeManager, flagManager, storeUtilsProxy, aptoideAccountManager, appViewConfiguration,
+            preferencesManager, downloadStateParser, appViewAnalytics, notificationAnalytics,
+            installAnalytics, limit, socialRepository);
+
+    when(appCenter.loadDetailedApp((long) 1, "anyString", "anyString")).thenReturn(
+        Single.just(detailedAppRequestResult));
+    when(storeManager.isSubscribed(anyLong())).thenReturn(Observable.just(true));
+
+    appViewManager.loadAppViewViewModel()
```

Comment:
No need to do the toBlocking() and value() call if it is not being saved and used.